modules/hal_gigadevice/CMakeLists.txt

```diff
@@ -6,20 +6,34 @@ if(CONFIG_HAS_GD32_HAL)
 string(TOUPPER ${CONFIG_SOC} gd32_soc_uc)
 
 set(gd32_soc_dir     ${ZEPHYR_HAL_GIGADEVICE_MODULE_DIR}/${CONFIG_SOC_SERIES})
-set(gd32_cmsis_dir   ${gd32_soc_dir}/cmsis/gd/${CONFIG_SOC_SERIES})
+if(CONFIG_SOC_FAMILY_GD32_ARM)
+  set(gd32_soc_sys_dir   ${gd32_soc_dir}/cmsis/gd/${CONFIG_SOC_SERIES})
+elseif(CONFIG_SOC_SERIES_GD32VF103)
+  set(gd32_soc_sys_dir   ${gd32_soc_dir}/riscv)
+  zephyr_include_directories(${gd32_soc_dir}/riscv/drivers)
+endif()
+
 set(gd32_std_dir     ${gd32_soc_dir}/standard_peripheral)
 set(gd32_std_src_dir ${gd32_std_dir}/source)
 
 zephyr_library_named(hal_gigadevice)
 
 zephyr_compile_definitions(${gd32_soc_uc})
 
+if(${CONFIG_GD32_HXTAL_8MHZ})
+  zephyr_compile_definitions(HXTAL_VALUE=8000000)
+  zephyr_compile_definitions(HXTAL_VALUE_25M=HXTAL_VALUE)
+elseif(${CONFIG_GD32_HXTAL_25MHZ})
+  zephyr_compile_definitions(HXTAL_VALUE=25000000)
+  zephyr_compile_definitions(HXTAL_VALUE_8M=HXTAL_VALUE)
```

Comment:
8M/25M are mixed now.

I think you need:

```
zephyr_compile_definitions(HXTAL_VALUE=8000000 HXTAL_VALUE_8M=8000000)
```

and

```
zephyr_compile_definitions(HXTAL_VALUE=25000000 HXTAL_VALUE_25M=25000000)
```

Reply:
In gd32vf103 case, HXTAL_VALUE_8M and HXTAL_VALUE_25M is defined here.

https://github.com/zephyrproject-rtos/hal_gigadevice/blob/2eafc9d95b623757df0027dd08ed1b693d3bb770/gd32vf103/riscv/include/gd32vf103.h#L59

```
 #if !defined  HXTAL_VALUE  
   #ifdef GD32VF103R_START
   #define HXTAL_VALUE    ((uint32_t)25000000) /*!< value of the external oscillator in Hz */
   #define HXTAL_VALUE_8M  HXTAL_VALUE
#elif defined(GD32VF103V_EVAL) || defined(GD32VF103C_START) || defined(GD32VF103T_START)
   #define HXTAL_VALUE    ((uint32_t)8000000) /*!< value of the external oscillator in Hz */
   #define HXTAL_VALUE_25M  HXTAL_VALUE
   #else
   #error "Please select the target board type used in your application (in gd32vf103.h file)"
   #endif
 #endif /* high speed crystal oscillator value */
```

HXTAL_VALUE_8M and HXTAL_VALUE_8M expand to HXTAL_VALUE that is at immediately before.
That code expands to the following.

```
 #if !defined  HXTAL_VALUE  
   #ifdef GD32VF103R_START
   #define HXTAL_VALUE    ((uint32_t)25000000) /*!< value of the external oscillator in Hz */
   #define HXTAL_VALUE_8M  ((uint32_t)25000000)
#elif defined(GD32VF103V_EVAL) || defined(GD32VF103C_START) || defined(GD32VF103T_START)
   #define HXTAL_VALUE    ((uint32_t)8000000) /*!< value of the external oscillator in Hz */
   #define HXTAL_VALUE_25M   ((uint32_t)8000000)
...
```

Therefore, I think it should be define as below.

```
zephyr_compile_definitions(HXTAL_VALUE=8000000 HXTAL_VALUE_25M=8000000)
```
and
```
zephyr_compile_definitions(HXTAL_VALUE=25000000 HXTAL_VALUE_8M=25000000)
```

Once, I push with it.

(In fact, HXTAL_VALUE_25M and HXTAL_VALUE_8M is not referenced in GigaDevice Firmware sources, it does not cause any problems.
I think it is wrond. gd32e10x's definitions (HXTAL_VALUE=8000000 HXTAL_VALUE_8M=8000000) perhaps correct.
But we can't determine also the definition is wrong or not because it is not used.

HXTAL_VALUE_25M and HXTAL_VALUE_8M should keep source behavior if defining, I think.)